addons/blocks2image/userscript.js

```diff
@@ -0,0 +1,329 @@
+    blocksMedia.set("dropdown-arrow.svg", "data:image/svg+xml;base64,PHN2ZyBpZD0iTGF5ZXJfMSIgZGF0YS1uYW1lPSJMYXllciAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxMi43MSIgaGVpZ2h0PSI4Ljc5IiB2aWV3Qm94PSIwIDAgMTIuNzEgOC43OSI+PHRpdGxlPmRyb3Bkb3duLWFycm93PC90aXRsZT48ZyBvcGFjaXR5PSIwLjEiPjxwYXRoIGQ9Ik0xMi43MSwyLjQ0QTIuNDEsMi40MSwwLDAsMSwxMiw0LjE2TDguMDgsOC4wOGEyLjQ1LDIuNDUsMCwwLDEtMy40NSwwTDAuNzIsNC4xNkEyLjQyLDIuNDIsMCwwLDEsMCwyLjQ0LDIuNDgsMi40OCwwLDAsMSwuNzEuNzFDMSwwLjQ3LDEuNDMsMCw2LjM2LDBTMTEuNzUsMC40NiwxMiwuNzFBMi40NCwyLjQ0LDAsMCwxLDEyLjcxLDIuNDRaIiBmaWxsPSIjMjMxZjIwIi8+PC9nPjxwYXRoIGQ9Ik02LjM2LDcuNzlhMS40MywxLjQzLDAsMCwxLTEtLjQyTDEuNDIsMy40NWExLjQ0LDEuNDQsMCwwLDEsMC0yYzAuNTYtLjU2LDkuMzEtMC41Niw5Ljg3LDBhMS40NCwxLjQ0LDAsMCwxLDAsMkw3LjM3LDcuMzdBMS40MywxLjQzLDAsMCwxLDYuMzYsNy43OVoiIGZpbGw9IiNmZmYiLz48L3N2Zz4=")
+
+
+    let svg = document.createElement('svg')
+    svg.setAttribute('xmlns', 'http://www.w3.org/2000/svg')
+    svg.setAttribute('xmlns:html', 'http://www.w3.org/1999/xhtml')
+    svg.setAttribute('xmlns:xlink', 'http://www.w3.org/1999/xlink')
+    svg.setAttribute('version', '1.1')
+
+    let style = document.createElement('style')
+    style.textContent =
+        `
+.blocklyText {
+    fill: #fff;
+    font-family: "Helvetica Neue", Helvetica, sans-serif;
+    font-size: 12pt;
+    font-weight: 500;
+}
+.blocklyNonEditableText>text, .blocklyEditableText>text {
+    fill: #575E75;
+}
+.blocklyDropdownText {
+    fill: #fff !important;
+}
+`
+    if (request.command === 'export1' || request.command === 'export3') {
+        svg = selectedBlocks(svg, style, isExportPNG)
+    } else {
+        svg = allBlocks(svg, style, isExportPNG)
+    }
+    // resolve nbsp whitespace
+    let texts = Array.from(svg.getElementsByTagName('text'))
+    texts.forEach(text => {
+        text.innerHTML = text.innerHTML.replace(/&nbsp;/g, ' ')
```

Comment:
You can't use innerHTML

Reply:
No, that should be fine. It's not dynamic, so it's un-xssable.